File: 2018/day05/day05_test.go
package main

import (
	"testing"
)

func TestPart1(t *testing.T) {
	lines := readFile("day05.test.txt")

	expected := 10
	actual := part1(lines)
	if expected != actual {
		t.Fatalf("Expected %v but was %v", expected, actual)
	}
}

func TestPart2(t *testing.T) {
	lines := readFile("day05.test.txt")

	expected := 4
	actual := part2(lines)
	if expected != actual {
		t.Fatalf("Expected %v but was %v", expected, actual)
	}
}

File: solutions/2025/day05/tests/day05_test.py
from day05.day05 import Day05
import unittest


class TestDay05(unittest.TestCase):
    def test_part1(self):
        input = Day05().read_file("day05/tests/day05.test.txt")
        self.assertEqual(Day05.solve_part1(input), 3)

    def test_part2(self):
        input = Day05().read_file("day05/tests/day05.test.txt")
        self.assertEqual(Day05.solve_part2(input), 14)


if __name__ == "__main__":
    unittest.main()
